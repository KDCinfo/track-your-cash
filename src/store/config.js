const config = {
    homeText: 'Home',
    loginText: 'Login',
    registerText: 'Register',
    aboutText: 'About',
    readmeText: 'Read-Me',
    dateText: 'dateText',
    amountText: 'amountText',
    descriptionText: 'descriptionText',
    typeText: 'typeText',
    categoryText: 'categoryText',
    notesText: 'notesText',
    reconciledText: 'reconciledText',
    dateFormat: 'mmmm dS, yyyy - h:MM:ss TT',
    dollarAmountRegex: '(?=.)^(([1-9][0-9]{0,2}(,[0-9]{3})*)|[0-9]+)?(.[0-9]{1,2})?$',
    pagination: 10
};

    // Linter complained about the \ (near the end)
    // dollarAmountRegex : '(?=.)^(([1-9][0-9]{0,2}(,[0-9]{3})*)|[0-9]+)?(\.[0-9]{1,2})?$',

export default config;
