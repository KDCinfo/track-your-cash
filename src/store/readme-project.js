const myMarkP = `# Project Notes

This entire notes section is \`coded in Markdown\` for a little easier prettification (as opposed to formatting it with full HTML tagging).

## App State

\`\`\`
  loggedInId: '',
  base: 0,
  registry: [
    {
      id, date, description, amount, type, category, notes, reconciled
    }
  ],
  visibilityFilter: SHOW_ALL,
  withdrawError : false,
  transactionSuccess : false,
\`\`\`

## Components

### App

  - Header
  - Content*
  - Footer

### Content - Login

('Get Started' page)

  - E-mail
  - [Login] (An account will be created if e-mail does not already exist)

### Content - Registry

  Entry Date | Entry Type | Entry Description | -(Entry Amount) | [Update]

  Category   | Notes                          | Reconciled      | [Delete]

  Add Entry                                                     | [Logout]
`

export default myMarkP