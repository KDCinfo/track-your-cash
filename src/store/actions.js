import {actionCreator, getLastTransaction, getObjectFromStorage} from './functions'

/*
 * action types
 */
export const ADD_TODO = 'ADD_TODO'
export const TOGGLE_TODO = 'TOGGLE_TODO'
export const SET_VISIBILITY_FILTER = 'SET_VISIBILITY_FILTER'

export const CLEAR_FORM = 'CLEAR_FORM'
export const CLEAR_FORM_EXIST = 'CLEAR_FORM_EXIST'
export const INPUT_TYPING = 'INPUT_TYPING'
export const UPDATE_STATE_FIELD = 'UPDATE_STATE_FIELD'
export const ADD_TO_STATE_ARRAY = 'ADD_TO_STATE_ARRAY'
export const REMOVE_FROM_STATE_ARRAY = 'REMOVE_FROM_STATE_ARRAY'
export const UPDATE_REGISTRY_ENTRY = 'UPDATE_REGISTRY_ENTRY'
export const LOGIN_NEW = 'LOGIN_NEW'
export const LOGIN_EXISTING = 'LOGIN_EXISTING'
export const SUBMIT_TRANSACTION = 'SUBMIT_TRANSACTION'
export const INVALID_TRANSACTION = 'INVALID_TRANSACTION'
export const LOGOUT = 'LOGOUT'

/*
 * other constants
 */

export const VisibilityFilters = {
    SHOW_ALL: 'SHOW_ALL',
    SHOW_COMPLETED: 'SHOW_COMPLETED',
    SHOW_ACTIVE: 'SHOW_ACTIVE'
}

export const clearForm = actionCreator(CLEAR_FORM)
export const clearFormExist = actionCreator(CLEAR_FORM_EXIST)
export const inputTyping = actionCreator(INPUT_TYPING, 'key', 'value')
export const updateStateField = actionCreator(UPDATE_STATE_FIELD, 'key', 'value')
export const addToStateArray = actionCreator(ADD_TO_STATE_ARRAY, 'key', 'value')
export const removeFromStateArray = actionCreator(REMOVE_FROM_STATE_ARRAY, 'previousEntry')
export const updateRegistryEntry = actionCreator(UPDATE_REGISTRY_ENTRY, 'updatedEntry')
const logout = actionCreator(LOGOUT)
const loginNew = actionCreator(LOGIN_NEW, 'email')
const loginExisting = actionCreator(LOGIN_EXISTING, 'email', 'loadedState')
const invalidTransaction = actionCreator(INVALID_TRANSACTION)
const submitTransactionValid = actionCreator(SUBMIT_TRANSACTION, 'whichType', 'amount', 'description', 'date')

/*
 * action creators
 */

export function addTodo(text) {
    return { type: ADD_TODO, text }
}

export function toggleTodo(index) {
    return { type: TOGGLE_TODO, index }
}

export function setVisibilityFilter(filter) {
    return { type: SET_VISIBILITY_FILTER, filter }
}

export const logoutUser = (LOGOUT) => {
    console.log('logging out...')
    return dispatch => {
        dispatch(logout())
    }
}

export const login = email => {
    return dispatch => {
        let item = getObjectFromStorage(localStorage, email)
        if(item){
            dispatch(loginExisting(email, item))
        } else {
            dispatch(loginNew(email))
        }
    }
}

export const submitTransaction = (whichType, amount, description, date) => {
    return (dispatch, getState) => {
        let lastTransaction = getLastTransaction(getState().transactionList, getState().transactionEntries)
        let balance = lastTransaction ? lastTransaction.balance : 0
        if(balance < amount) {
            dispatch(invalidTransaction())
        } else {
            dispatch(submitTransactionValid(whichType, amount, description, date))
        }
    }
}
