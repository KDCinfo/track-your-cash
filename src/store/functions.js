import formatDate from 'dateformat';
import config from '../store/config';
import * as ACTIONS from '../store/actions';

export function actionCreator(type, ...argNames) {
    return function(...args) {
        let action = {
            type
        }
        argNames.forEach((arg, index) => {
            action[argNames[index]] = args[index];
        });
        return action;
    }
}

export const formatCurrency = amount => {
    let amt = amount;
    let formatted = '';
    if(amt < 0) {
        formatted += '-';
        amt *= -1;
    }
    formatted += '$';
    formatted += amt.toFixed(2);
    return formatted;
};

export const getFormattedDate = date => {
    return formatDate(date, config.dateFormat);
}

export const getLastTransactionId = tArr => {
    return tArr[0];
}

export const getLastTransaction = (tArr, transactions) => {
    const lastId = getLastTransactionId(tArr);
    return transactions[lastId];
};

export const setStorageItem = (storage, key, value) => {
    try {
        storage.setItem(key, value);
    } catch(e) {
        console.error(e);
    }
}

export const getStorageItem = (storage, key) => {
    try {
        return storage.getItem(key);
    } catch(e) {
        console.error(e);
        return null;
    }
}

export const getObjectFromStorage = (storage, key) => {
    let sItem = getStorageItem(storage, key);
    if(sItem !== undefined) {
        sItem = JSON.parse(sItem);
    }
    return sItem;
}

export const deleteStorageItem = (storage, key) => {
    try {
        storage.removeItem(key);
    } catch(e) {
        console.error(e);
    }
}

export const localStorageMiddleWare = store => next => action => {
    let result = next(action);
    if(action.type === ACTIONS.LOGIN_NEW) {
        let newState = store.getState();
        let {registry} = newState;
        let dataObj = {registry};
console.log('localStorageMiddleWare LOGIN_NEW', newState, dataObj, JSON.stringify(dataObj))
        setStorageItem(localStorage, newState.loggedInId, JSON.stringify(newState));
    }
    if(action.type === ACTIONS.SUBMIT_TRANSACTION) {
        let newState = store.getState();
        let {registry} = newState;
        let dataObj = {registry};
console.log('localStorageMiddleWare SUBMIT_TRANSACTION', newState, dataObj, JSON.stringify(dataObj))
        setStorageItem(localStorage, newState.loggedInId, JSON.stringify(dataObj));
    }
    return result;
}
