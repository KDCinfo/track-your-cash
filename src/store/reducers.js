import * as ACTIONS from './actions'
// import {getLastTransaction} from './functions'
import initialState, { getLoadedEntry } from './initial-state'

const clearFormObject = () => {
    return {
        withdrawError: false,
        transactionSuccess: false,
        currentEntry: getLoadedEntry()
    }
}
const clearFormExistObject = () => {
    return {
        withdrawError: false,
        transactionSuccess: false,
        existingEntry: {}
    }
}
/*
  const buildTransactionObject = (type, amount, description, date) => {
      const amt = amount
      const newBalance = amt
      return {
          date: date,
          description: description,
          amount: amt,
          balance: newBalance
      }
  }
  const getNewStateAfterTransaction = (date, entry, entries) => {
      const newEntries = {...entries, [date]: entry}
      const obj = {transactionEntries: newEntries, transactionSuccess: true, pageIndex: 1}
      return {...clearFormObject(), ...obj}
  }
*/

const registerReducer = (state = {}, action) => {
    switch (action.type) {
        case ACTIONS.CLEAR_FORM:
          return {...state, ...clearFormObject()}
        case ACTIONS.CLEAR_FORM_EXIST:
          return {...state, ...clearFormExistObject()}
        case ACTIONS.INPUT_TYPING: {
          const obj = {[action.key]: action.value}
          return {...state, ...obj}
        }
        case ACTIONS.UPDATE_STATE_FIELD: {
          const obj = {[action.key]: action.value}
          return {...state, ...obj}
        }
        case ACTIONS.ADD_TO_STATE_ARRAY: {

//           const oldStateArray = state[action.key]
// console.log('[reducers 5a]', oldStateArray)

//           let newStateArray;
//                 oldStateArray.push(action.value)
//                 newStateArray = oldStateArray


          const oldStateArray = state[action.key],
                newRegistryEntry = action.value
console.log('[NEW - oldStateArray 0]', typeof(oldStateArray), Object.prototype.toString.call( oldStateArray ), oldStateArray)
          let newStateArray;
                newStateArray = oldStateArray.concat([newRegistryEntry])

                // obj = {[action.key]: newState}
console.log('[reducers 1]', typeof(action.key), action.key)   // string, registry
console.log('[reducers 2]', action.value) // JSON parsed object
console.log('[reducers 3a]', state[action.key])
console.log('[reducers 3b]', newStateArray)     // [Object, Object]
// console.log('[reducers 4]', obj)
          // JSON.stringify({...oldState, registry: newRegistry })
          return {...state, [action.key]: newStateArray}
          // return {...state}
        }
        case ACTIONS.INPUT_TYPING_REG: {
          const registryObj = [],
                obj = {withdrawError: false, transactionSuccess: false, registry: registryObj.concat( { [action.key]: action.value } ) }
          return {...state, ...obj}
        }
        case ACTIONS.LOGIN_NEW: {
          const newState = {...initialState, loggedInId: action.email}
          return {...state, ...newState}
        }
        case ACTIONS.LOGIN_EXISTING: {
          // loggedInId: action.payload
          const newState = {...action.loadedState, loggedInId: action.email}
          return {...state, ...newState}
        }
        case ACTIONS.LOGOUT:
          return {...state, ...initialState}
        case ACTIONS.INVALID_TRANSACTION:
          return {...state, withdrawError: true}
        case ACTIONS.SUBMIT_TRANSACTION: {
          // const entry = buildTransactionObject(action.whichType, action.amount, action.description, action.date, state);
          // const newState = getNewStateAfterTransaction(state.transactionList, action.date, entry, state.transactionEntries)
          // return {...state, ...newState}
          return {...state}
        }
        case ACTIONS.ADD_TODO:
          return [
            ...state,
            {
              id: state.length,
              date: new Date(),
              description: action.description,
              amount: action.amount,
              type: action.type,
              category: action.category,
              notes: action.notes,
              reconciled: false
            }
          ]
        case ACTIONS.TOGGLE_TODO:
          return state.map((todo, index) => {
            if (index === action.index) {
              return Object.assign({}, todo, {
                reconciled: !todo.reconciled
              })
            }
            return todo
          })
        default:
          return state
    }
}

export default registerReducer
