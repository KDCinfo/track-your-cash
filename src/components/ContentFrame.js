import React from 'react'
import { connect } from 'react-redux'
import { withRouter } from 'react-router-dom'
import {bindActionCreators} from 'redux';

import { NavLink } from 'react-router-dom'
import { PageHeader } from 'react-bootstrap'

import Logout from './Logout'
import { getStorageItem, deleteStorageItem } from '../store/functions'
import * as ACTIONS from '../store/actions'

class ContentFrameContainer extends React.Component {
    constructor(props) {
        super(props)

        const isLocal = window.location.href.indexOf('192.168') > 0

        this.handleClickLogout = this.handleClickLogout.bind(this)
        this.getWhichLink = this.getWhichLink.bind(this)

        this.state = {
            showCRA: isLocal ?
                <span>
                    [ <NavLink to="/cra">CRA</NavLink> ]&nbsp;
                </span>
                : ''
        }
    }
    getWhichLink() {
        const loggedInIdLocal = getStorageItem(sessionStorage, 'user') || '',
              isLoggedIn = loggedInIdLocal.length > 0

        return (
            isLoggedIn ?
                <span>
                    [ <NavLink to="/register">Your Register</NavLink> ]&nbsp;
                    [ <span className="NavLink"><Logout handleClick={this.handleClickLogout} /></span> ]&nbsp;
                </span>
                :
                <span>
                    [ <NavLink to="/login">Get Started</NavLink> ]&nbsp;
                </span>
        )
    }
    handleClickLogout() {
        this.props.actions.logoutUser();
        deleteStorageItem(sessionStorage, 'user');
        deleteStorageItem(sessionStorage, 'entry');
        this.props.history.push('/');
    }
    render() {
        return (
            <div>
                <header>
                    <PageHeader>
                        Track Your Cash<br/>
                        <small>(a checking register)</small>
                    </PageHeader>
                    [ <NavLink exact to="/">Home</NavLink> ]&nbsp;
                    {this.getWhichLink()}
                    [ <NavLink to="/about">About</NavLink> ]&nbsp;
                    [ <NavLink to="/readme">Project Notes</NavLink> ]&nbsp;
                    {this.state.showCRA}
                </header>

                <main className="content">{this.props.children}</main>

                <footer>Footer</footer>
            </div>
        )
    }
}

const mapStateToProps = state => {
    return {
        loggedInId: state.loggedInId
    }
}

const mapDispatchToProps = dispatch => {
    return {
        actions: bindActionCreators(ACTIONS, dispatch)
    }
}

const ContentFrame = withRouter(connect(
    mapStateToProps,
    mapDispatchToProps
)(ContentFrameContainer))

export default ContentFrame