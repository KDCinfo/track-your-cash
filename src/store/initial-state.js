import { getStorageItem, getObjectFromStorage, setStorageItem, deleteStorageItem } from './functions'

const initialState = {
    loggedInId: '',
    base: 0,
    registry: [],
    currentEntry: {},
    types: ['Deposit', 'Withdraw', 'EFT', 'Cash', 'ATM', 'Debit', 'Credit'],
    categorys: [
        'Income - Salary',
        'Income - Refund',
        'Expense - Dining',
        'Expense - Entertainment',
        'Expense - Groceries',
        'Expense - Health',
        'Expense - Petty Cash',
        'Expense - Travel',
    ]
}

const newDate = new Date()

const currentEntry = {
    "id": 0,
    "date": (newDate.getFullYear()) + ('0' + (newDate.getMonth()+1)).slice(-2) + ('0' + newDate.getDate()).slice(-2),
    "description": "",
    "amount": "0",
    "type": "",
    "category": "",
    "notes": "",
    "reconciled": false
}

export const newCurrentEntry = {
    "description": "",
    "amount": "0",
    "type": "",
    "category": "",
    "notes": "",
    "reconciled": false
}

// INITIAL LOAD ORDER

// const initialStateLoad = {...initialState, ...getLoadedState(), currentEntry: getLoadedEntry()}

export const getLoadedState = () => {
    let loadedState = {},
        userId = getStorageItem(sessionStorage, 'user')

    if(userId) {
        let getLocalUserStorage = getObjectFromStorage(localStorage, userId)

        if (getLocalUserStorage) {
            loadedState.loggedInId = userId
            loadedState.registry = getLocalUserStorage.registry
        } else {
            // Session 'user' exists, but no localStorage 'user' found
            // Likely a problem when creating localStorage: Start over
            loadedState.loggedInId = ''
            deleteStorageItem(sessionStorage, 'user')
            deleteStorageItem(sessionStorage, 'entry')
            deleteStorageItem(sessionStorage, 'entryExist')
        }
    }
    return {...initialState, ...loadedState}
}

export const getLoadedEntry = () => {
    const userObjStor = getStorageItem(sessionStorage, 'user'),
          entryObjStor = getStorageItem(sessionStorage, 'entry')

    let entryObj

    if(!entryObjStor) {                             // If no session entry, create it
        entryObj = currentEntry                     // create it using [currentEntry] (set above)
        if(userObjStor) {                           // but only if there is an active user
            setStorageItem(sessionStorage, 'entry', JSON.stringify(entryObj));
        }                                           // else just return an initialized (empty) entry object
    } else {
        entryObj = JSON.parse(entryObjStor)         // If session entry exists, return it for initializing state
    }
    return entryObj
}

export const getLoadedEntryExisting = () => {       // I'm unsure I'm using this (yet)
    const userObjStor = getStorageItem(sessionStorage, 'user'),
          entryObjStor = getStorageItem(sessionStorage, 'entryExist')

    let entryObj

    if(!entryObjStor) {
        entryObj = currentEntry
        if(userObjStor) {
            setStorageItem(sessionStorage, 'entryExist', JSON.stringify(entryObj));
        }
    } else {
        entryObj = JSON.parse(entryObjStor)
    }
    return entryObj
}

export default initialState
/*
    [index.js]
        Initial Load
            -> Set State: initialStateLoad = {
                ...initialState,
                ...getLoadedState(),
                currentEntry: getLoadedEntry() }

    [Login.js]                      -> handleSubmit()
        -> Run getLoadedEntry()

    [Register.js]
        Initial Load
            -> Run getLoadedEntry()
            -> mapStateToProps      -> state.currentEntry

    [RegisterEntryFilled.js]
        Add Form: Update Fields     -> handleChangeAmount() -> handleStorageSet()
            -> setStorageItem(sessionStorage, 'entry', JSON.stringify(entryObjNew))
            -> this.props.actions.inputTyping('currentEntry', entryObjNew)

        Add to Register
            -> Update (increment) Register ID (currentEntry.Id)
            -> Copy currentEntry    -> state.registry[]
            -> Clear sessionStorage 'entry'
            -> Run getLoadedEntry()

        Existing Form: Update Fields
            -> [Edit] button
                -> Add selected existing register entry to sessionStorage['existingEntry']
                ...These should happen automatically when 'existingEntry' is populated
                    -> Show [Save] buttton + [Cancel] button (hide [Delete] button)
                    -> Enable selected existing register entry

        Update Register
            -> Get current Register ID (Key: entry.Id)
            -> Copy 'existingEntry' to replace existing state.registry[]
            -> Clear sessionStorage['existingEntry']

    [ContentFrame.js]
        Logout Nav Link             -> handleClickLogout()
            -> Set State: initialState = { loggedInId: '', registry: [], currentEntry: {} }
            -> Remove sessionStorage 'user'
            -> Remove sessionStorage 'entry'

    -> getLoadedEntry = () =>
        -> Check for storage
            None
                FUNCTION
                    -> Add currentEntry initial object to storage
                REDUCER
                    -> Add currentEntry initial object to state
            Exists
                REDUCER
                    -> Add/Overwite state with storage {...state, ...storageEntry}

    [loggedInId: '']
    [registry: [ {}, {}, {} ]]
    [base: 0]

    // 'Base' is to allow for future archiving
        // The live register base (starting total) will be the totality of the archived registries

    registry: [
        {   "id" : 0,
            "date": "",
            "description": "",
            "amount": 0,
            "type": "",
            "category": "",
            "notes": "",
            "reconciled": false
        }
    ]
    registry: [
        {   "id" : 0,
            "date": 'Today',
            "description": "First",
            "amount": 1.25,
            "type": "Deposit",
            "category": "Household",
            "notes": "Initializing an entry",
            "reconciled": false
        }
    ]
*/