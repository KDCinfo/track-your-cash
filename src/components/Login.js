import React from 'react'
import { connect } from 'react-redux'
import { withRouter } from 'react-router-dom'
import {bindActionCreators} from 'redux';

import { Well, Grid, Row, Col } from 'react-bootstrap'

import LoginForm from './LoginForm'
import * as ACTIONS from '../store/actions';
import {setStorageItem} from '../store/functions';
import config from '../store/config';
import { getLoadedEntry } from '../store/initial-state'

class LoginContainer extends React.Component {
    constructor(props) {
        super(props)
        this.state = {
            inputMessage: (typeof(this.props.inputMessage) === 'undefined' || this.props.inputMessage.length === 0)
                ? 'E-mail (a valid e-mail address is required)'
                : this.props.inputMessage
        }
    }
    componentWillUnmount() {
        this.props.actions.clearForm();
        this.props.actions.clearFormExist();
    }
    handleChangeLogin = e => {
        this.props.actions.inputTyping('loggedInId', e.target.value);
    }
    handleSubmit = e => {
        e.preventDefault();
        const email = this.props.loggedInId

        this.props.actions.login(email)
        setStorageItem(sessionStorage, 'user', email)
        this.props.actions.updateStateField('currentEntry', getLoadedEntry())
        this.props.history.push('/' + config.registerText)
    }
    render() {
        const {loggedInId} = this.props;
        const handleChangeLogin = this.handleChangeLogin;
        const handleSubmit = this.handleSubmit;
        const inputMessage = this.state.inputMessage
        const propsObj = {loggedInId, handleChangeLogin, handleSubmit, inputMessage};
        return (
            <Grid fluid={true}>
                <Row>
                    <Col xs={12}>
                        <section>
                            Enter your e-mail below to login or create an account.
                        </section>
                    </Col>
                </Row>
                <Well bsSize="large">
                    <LoginForm {...propsObj} />
                </Well>
                <Row>
                    <Col xs={12}>
                        <aside>
                            An account will be created if the provided e-mail does not already exist in the system.
                            This is a public 'for fun' register and should not be used to keep track of your real finances.
                            You can, however, <a href="https://github.com/KDCinfo/track-your-cash" target="kdcNewWindow">fork this repo from GitHub</a> and set it up on your own local box, or somewhere behind authentication.
                        </aside>
                    </Col>
                </Row>
            </Grid>
        )
    }
}

const mapStateToProps = state => {
    return {
        loggedInId: state.loggedInId
    }
}

const mapDispatchToProps = dispatch => {
    return {
        actions: bindActionCreators(ACTIONS, dispatch)
    }
}

const Login = withRouter(connect(
    mapStateToProps,
    mapDispatchToProps
)(LoginContainer))

export default Login
