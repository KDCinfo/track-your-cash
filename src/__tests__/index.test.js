/*
    I have very little knowledge of testing,
    and have no idea if including everything as I have below
    is correct or not (as it's done in the main [index.js] file).

    Unsure how routing is tested; if perhaps routes are entered manually,
    or if there'd be a suite of imports covering each route from separate test files.

    But... the below passed the 2 core tests:
        // Do snapshots align and measure up?
        // Plain and simple --> Does it load?
*/
import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import { createStore, applyMiddleware, compose } from 'redux'
import { Provider } from 'react-redux'
import thunkMiddleware from 'redux-thunk';

import { Router, Route, Switch } from 'react-router'
import createBrowserHistory from 'history/createBrowserHistory'

import { localStorageMiddleWare } from '../store/functions';

import reducers from '../store/reducers'

import initialState from '../store/initial-state'

import ContentFrame from '../components/ContentFrame'
import Login from '../components/Login'

    const inputMessage = 'E-mail (required)'
    const LoginContainer = () => <ContentFrame><Login inputMessage={inputMessage} /></ContentFrame>

    const initialStateLoad = () => initialState
    const customHistory = createBrowserHistory()

let middleware = [thunkMiddleware, localStorageMiddleWare];

    if(process.env.NODE_ENV !== 'production') {
        let logger = require('redux-logger');
        const loggerMiddleware = logger.createLogger();
        middleware = [...middleware, loggerMiddleware];
    }

let store = createStore(reducers, initialStateLoad(
    applyMiddleware(...middleware)
))

const App = ({ store }) => (
    <Provider store={store}>
        <Router history={customHistory}>
            <Route path="/" component={LoginContainer} />
        </Router>
    </Provider>
)

// The tests included below, although they do pass, are incomplete
// and meant only to serve as a base for testing (PRs are welcome)

    // Tests Include:
    // Jest Snapshots
    // Does it Crash?

// This article is a good resource for component testing
// https://www.sitepoint.com/test-react-components-jest/

// [describe] Optional - For logical grouping

// Can wrap Tests in either [test] or [it]

describe('LoginContainer:', () => {

    test('Jest Snapshot', () => {
        const component = renderer.create(
            <App store={store} />
        )
        let tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<App store={store} />, div)
    })
})