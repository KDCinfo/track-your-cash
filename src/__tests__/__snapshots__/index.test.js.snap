// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginContainer: Jest Snapshot 1`] = `
<div
  className="div-c"
>
  <header>
    <div
      className="page-header"
    >
      <h1>
        Track Your Ca
        <small>
          <b>
            $
          </b>
        </small>
        h
        <br />
        <small>
          (a checking register)
        </small>
      </h1>
    </div>
    <nav>
      <span
        className="no-wrap"
      >
        <a
          className="active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Home
        </a>
      </span>
       
      <span>
        <span
          className="no-wrap"
        >
          <a
            className={undefined}
            href="/login"
            onClick={[Function]}
            style={undefined}
          >
            Get Started
          </a>
        </span>
         
      </span>
      <span
        className="no-wrap"
      >
        <a
          className={undefined}
          href="/readme"
          onClick={[Function]}
          style={undefined}
        >
          Project Notes
        </a>
      </span>
       
      <span>
        <span
          className="no-wrap"
        >
          <a
            className={undefined}
            href="/cra"
            onClick={[Function]}
            style={undefined}
          >
            CRA
          </a>
        </span>
         
      </span>
      <span
        className="no-wrap"
      >
        <a
          className={undefined}
          href="/about"
          onClick={[Function]}
          style={undefined}
        >
          About
        </a>
      </span>
       
      <span />
    </nav>
  </header>
  <main
    className="content"
  >
    <div
      className="container-fluid"
    >
      <div
        className="row"
      >
        <div
          className="col-xs-12"
        >
          <section>
            <h3>
              Access Your Register
            </h3>
            <p>
              Enter your e-mail below to login or create an account.
            </p>
          </section>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-xs-12"
        >
          <main>
            <div
              className="well well-lg"
            >
              <form
                className="login-form "
                onSubmit={[Function]}
              >
                <div
                  className="show-grid row"
                >
                  <div
                    className="col-sm-7 col-xs-12"
                  >
                    <div
                      className="form-group"
                    >
                      <span
                        className="input-group"
                      >
                        <span
                          className="input-group-addon"
                        >
                          @
                        </span>
                        <input
                          autoFocus={true}
                          className="form-control"
                          id={undefined}
                          onChange={[Function]}
                          placeholder="E-mail (required)"
                          required={true}
                          title="E-mail (required)"
                          type="email"
                          value=""
                        />
                      </span>
                    </div>
                  </div>
                  <div
                    className="col-sm-5 col-xs-12"
                  >
                    <button
                      className="btn btn-default"
                      disabled={false}
                      type="submit"
                    >
                      Your Register Awaits...
                    </button>
                  </div>
                </div>
              </form>
            </div>
          </main>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-xs-12"
        >
          <aside>
            An account will be created if the provided e-mail does not already exist in the system. This is a public 'for fun' register and should not be used to keep track of your real finances. You can, however, 
            <a
              href="https://github.com/KDCinfo/track-your-cash"
              target="kdcNewWindow"
            >
              fork this repo from GitHub
            </a>
             and set it up on your own local box, or somewhere behind authentication.
          </aside>
        </div>
      </div>
    </div>
  </main>
  <footer>
    <div
      className="github"
    >
      Track Your Cash is 
      <a
        href="https://github.com/KDCinfo/track-your-cash"
        target="kdcNewWin"
      >
        Open Source on GitHub
      </a>
    </div>
    <div
      className="hide"
    >
      <button
        className="btn btn-sm btn-default"
        disabled={false}
        onClick={[Function]}
        title="Save your data to a local file you can view on your system."
        type="button"
      >
        Export (save your data)
      </button>
        
      <button
        className="btn btn-sm btn-default"
        disabled={false}
        onClick={[Function]}
        title="You will be asked to confirm the removal of all your data."
        type="button"
      >
        Delete Account
      </button>
    </div>
    <div
      className="footer-right"
    >
      <span
        className="mobile-only"
      >
        <span
          className="hide-created-575"
        >
          Created by: 
        </span>
        <a
          href="https://kdcinfo.com"
          target="kdcinfo"
        >
          KDC-Info
        </a>
      </span>
      <span
        className="non-mobile-i"
      >
        Created by: 
        <a
          href="https://kdcinfo.com"
          target="kdcinfo"
        >
          KDC-Info
        </a>
      </span>
    </div>
  </footer>
</div>
`;
